tensorboard/examples/plugins/example_basic/README.md

```diff
@@ -2,51 +2,159 @@
+
+If your plugin requires parsing flags or custom loading, consider defining a `TBLoader` as the entry point. Doing so is optional.
+
+For example:
+
+```python
+class MyLoader(base_plugin.TBLoader):
+  ### Plugin Setup
+
+  def define_flags(self, parser):
+    parser.add_argument_group('custom').add_argument('--enable_my_extras')
+
+  def fix_flags(self, flags):
+    if flags.enable_my_extras:
+      raise ValueError('Extras not ready')
+
+  def load(self, context):
+    return MyPlugin(context)
+```
+
+#### TBLoader
+  - `define_flags(parser)`: Optional method that takes an argparse.Namespace and exposes command-line flags. Please prefix flags with the name of the plugin to avoid collision.
+  - `fix_flags(flags)`: Optional method needed to fix or sanitize command-line flags.
+  - `load(context)`: Required method that takes a TBContext and returns a TBPlugin instance.
+
+It's recommended that plugins using flags call the `parser.add_argument_group(plugin_name)`. To learn more about the flag definition, see [docs](https://docs.python.org/library/argparse.html#adding-arguments)
+
+
+
+## Reading data from event files
+
+On instantiation, a plugin is provided a [`PluginEventMultiplexer`] object from which to read data. The `PluginRunToTagToContent` method on the multiplexer returns a dictionary containing all run–tag pairs and associated summary metadata for your plugin.
 
 Plugins are not technically restricted from arbitrary file system and network access, but we strongly recommend using the multiplexer exclusively. This abstracts over the filesystem (local or remote), provides a consistent user experience for runs and tags across plugins, and is optimized for TensorBoard read patterns.
 
 [`PluginEventMultiplexer`]: https://github.com/tensorflow/tensorboard/blob/master/tensorboard/backend/event_processing/plugin_event_multiplexer.py
 
+Example use of the multiplexer:
+```python
+class MyPlugin(base_plugin.TBPlugin):
+  def __init__(self, context):
+    self.multiplexer = context.multiplexer
+
+  def collect_data(self):
+    """
+    {runName: { images: [tag1, tag2, tag3],
+                scalarValues: [tagA, tagB, tagC],
+                histograms: [tagX, tagY, tagZ],
+                compressedHistograms: [tagX, tagY, tagZ],
+                graph: true, meta_graph: true}}
+    """
+    runs = self.multiplexer.Runs()
+
+    """
+    [
+      {wall_time: 100..., step: 1, tensor_proto: ...},
+      {wall_time: 100..., step: 2, tensor_proto: ...},
+      ...
+    ]
+    """
+    events = self.multiplexer.Tensors(run, tag)
+
+    """{run: {tag: content}, ...}"""
+    content = PluginRunToTagToContent(plugin_name)
+```
+
+For the complete EventMultiplexer API, see [`event_multiplexer.py`](https://github.com/tensorflow/tensorboard/blob/master/tensorboard/backend/event_processing/event_multiplexer.py).
```

Comment:
Please direct people to `plugin_event_multiplexer.py` instead. Also, perhaps mention DataProvider since multiplexer already has `deprecated` prefix to it.

Reply:
-> plugin_event_multiplexer.py: Done.

Documenting DataProvider's interface seems good to do eventually, but right now, `MultiplexerDataProvider`'s methods take `experiment_id`.  It seems like these signatures might be subject to change, and explaining what an `experiment` is, seems tricky to me.  Would it be reasonable to punt until the new `DataProvider` can replace the multiplexer for all data types?